executor/aggregate.go

```diff
@@ -862,6 +868,47 @@ func (e *HashAggExec) getPartialResults(groupKey string) []aggfuncs.PartialResul
 	return partialResults
 }
 
+func (e *HashAggExec) executeFirstRow(ctx context.Context, chk *chunk.Chunk) (err error) {
+	for {
+		if e.firstRowProcessed == e.childResult.NumRows() {
+			mSize := e.childResult.MemoryUsage()
+			e.firstRowProcessed = 0
+			err := Next(ctx, e.children[0], e.childResult)
+			e.memTracker.Consume(e.childResult.MemoryUsage() - mSize)
+			if err != nil {
+				return err
+			}
+			// no more data.
+			if e.childResult.NumRows() == 0 {
+				return nil
+			}
+			e.groupKeyBuffer, err = getGroupKey(e.ctx, e.childResult, e.groupKeyBuffer, e.GroupByItems)
+			if err != nil {
+				return err
+			}
+		}
+
+		failpoint.Inject("unparallelHashAggError", func(val failpoint.Value) {
+			if val.(bool) {
+				failpoint.Return(errors.New("HashAggExec.unparallelExec error"))
+			}
+		})
+
+		for e.firstRowProcessed < e.childResult.NumRows() {
+			groupKey := string(e.groupKeyBuffer[e.firstRowProcessed]) // do memory copy here, because e.groupKeyBuffer may be reused.
+			if !e.groupSet.Exist(groupKey) {
+				e.groupSet.Insert(groupKey)
+				e.groupKeys = append(e.groupKeys, groupKey)
+				chk.AppendRow(e.childResult.GetRow(e.firstRowProcessed))
```

Comment:
AppendRow directly maybe get wrong result.... for example,  
I create a table t (a int ,b int) and insert some values,
then run select distinct b,a from t. It will exchange the column, and return a wrong result.